kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -99,12 +105,12 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
           LOG.info("Writing to existing dataset/view: " + view);
           break;
         case CHECKPOINT:
-          boolean ready = true; // dataset.isReady(); // TODO: add when CDK-451 is ready
           long lastModForTarget = -1;
           if (view instanceof LastModifiedAccessor) {
             lastModForTarget = ((LastModifiedAccessor) view).getLastModified();
           }
-          if (ready && lastModForTarget > lastModForSource) {
+
+          if (ready && (lastModForTarget > lastModForSource)) {
```

Comment:
I think for this, the logic should still be in terms of the modified time. The ready signal updates that, so we get that the data was marked ready (after the source was modified) or the data was rewritten independent of the ready flag. I think we want that to be an or rather than requiring the flag. What if the data was written by something other than Kite and then consumed? There's no reason not to use it if its modified time is more recent.